paddle/fluid/operators/elementwise/mkldnn/elementwise_add_mkldnn_op.cc

```diff
@@ -125,46 +131,44 @@ class EltwiseAddMKLDNNKernel : public framework::OpKernel<T> {
       std::vector<int> dst_tz = framework::vectorize2int(z_dims);
 
       std::vector<memory::primitive_desc> srcs_pd;
-      std::vector<memory> srcs;
+      std::vector<std::shared_ptr<mkldnn::memory>> srcs;
```

Comment:
Please initialize the `srcs` with size 2 and replace push_backs with direct indexing intializations. Also consider using just a single vector instead of storing a vector shared pointers of memory and vector of the same memories.

Reply:
Good point. redundant container was removed